lib/matplotlib/colorbar.py

```diff
@@ -1158,22 +1157,22 @@ def _mesh(self):
         # vmax of the colorbar, not the norm.  This allows the situation
         # where the colormap has a narrower range than the colorbar, to
         # accommodate extra contours:
-        norm = copy.deepcopy(self.norm)
-        norm.vmin = self.vmin
-        norm.vmax = self.vmax
-        y, extendlen = self._proportional_y()
-        # invert:
-        if (isinstance(norm, (colors.BoundaryNorm, colors.NoNorm)) or
-                self.boundaries is not None):
-            y = y * (self.vmax - self.vmin) + self.vmin  # not using a norm.
-        else:
-            y = norm.inverse(y)
-        self._y = y
-        X, Y = np.meshgrid([0., 1.], y)
-        if self.orientation == 'vertical':
-            return (X, Y, extendlen)
-        else:
-            return (Y, X, extendlen)
+        with self.norm.callbacks.blocked(), \
```

Comment:
I guess maybe we should move this inside the `else` below?  Also why do you need to use setattr_cm here?  Maybe a comment?  

I guess I am still confused why we need a `deepcopy` here versus just a shallow `copy`.

Reply:
Good call on moving this to within the else. I just did that and added some more verbose comments.

The reason we don't want a copy is that there are signals attached to the norm to keep it in sync with the scalar mappable and autoscale_None machinery, so doing a temporary update would fire those signals and potentially propagate some unwanted changes or double update some features. The context manager is a nice way of doing the temporary update without needing to make a full copy.